src/models/BMF.jl

```diff
@@ -0,0 +1,55 @@
+"""
+$TYPEDEF 
+    BinaryMatrixFactorization{K}(A::AbstractMatrix{Bool}, k::Int)
+
+The Boolean Matrix Factorization (BMF) problem is defined on a binary matrix A in m x n size. Given a positive integer k, we need to determine whether we can factorize the matrix A into two binary matrices U and V such that the boolean product of U and V is equal to A, and the dimensions of U and V are (m x k) and (k x n) respectively. Refer to `Recent developments in boolean matrix factorization.(Miettinen, P., & Neumann, S,2020)` for details.
+
+### Required interfaces
+- [`variables`](@ref), the degrees of freedoms in the computational problem.
+- [`flavors`](@ref), the flavors (domain) of a degree of freedom.
+- [`solution_size`](@ref), the size (the lower the better) of the input configuration.
+- [`problem_size`](@ref), the size of the computational problem. e.g. for a graph, it could be `(n_vertices=?, n_edges=?)`.
+
+### Optional interfaces
+- [`num_variables`](@ref), the number of variables in the computational problem.
+- [`num_flavors`](@ref), the number of flavors (domain) of a degree of freedom.
+- [`findbest`](@ref), find the best configurations of the input problem.
+"""
+struct BinaryMatrixFactorization<: AbstractProblem
+    A::AbstractMatrix
```

Comment:
```suggestion
    A::BitMatrix
```

To avoid type instability.